pkg/api/validation/validation.go

```diff
@@ -628,6 +628,9 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource, fldPath *f
 	if len(hostPath.Path) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 	}
+
+	allErrs = append(allErrs, validatePathNoBacksteps(hostPath.Path, fldPath.Child("path"))...)
```

Comment:
@liggit Even without the apiserver validation, the kubelet would be performing the same checks retroactively on all host volumes. I don't see any easy way around that since, if i'm not mistaken, the kubelet does not have the applied psp for the pod. The kubelet only has the security context to go off of, which does not include any relevant information that would allow the kubelet to check backsteps only for pods with a PSP that has allowed host paths set. Thoughts?